src/transformers/generation_utils.py

```diff
@@ -1253,17 +1269,43 @@ def greedy_search(
             input_ids, max_length
         )
 
+        this_peer_finished = False
         while cur_len < max_length:
+
+            if synced_gpus:
```

Comment:
I like the code! 

One thing, I like to change is to **not** make the "normal" stopping criterion dependent on the variable `this_peer_finished`. IMO, most of the people reading this code will not use it in a `synced_gpus` setting and then the `this_peer_finished` variable logic is too complex IMO. I would prefer to make it very clear that `this_peer_finished` is only relevant when `synced_gpus` is True. Do you think we could do the following instead:

```python
            if unfinished_sequences.max() == 0 or stopping_criteria(input_ids, scores):
                #print(f"{dist.get_rank()} unfinished_sequences {cur_len}/{max_length}")
                if not synced_gpus:
                    break
                else:
                    this_peer_finished = True
```

            if not this_peer_finished and stopping_criteria(input_ids, scores):
                #print(f"{dist.get_rank()} hit stopping_criteria {cur_len}/{max_length}")
                this_peer_finished = True
```


a) Have all the "is-finished" code at the bottom. I think it should be possible to move this part of the code below

Reply:
Thank you for the review, @patrickvonplaten 

I like your suggestion - thank you!